### PR TITLE
Kraken

### PR DESCRIPTION
**Pull request by shershah010**
Adds Kraken as a broker. Note that Kraken only supports cryptos. Also only supports USD as currencies.

### CODE REVIEW

File: harvest/api/paper.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from harvest.utils import *
```
Comment (shershah010) [inline harvest/api/paper.py:13]: The only changes here are removing whitespace.

File: harvest/api/kraken.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+        'BAND': 'BAND',    'CTSI': 'CTSI',     'CQT': 'CQT',
+        'EWT': 'EWT',      'MLN': 'XMLN',      'ETH2': 'ETH2',
+        'GNO': 'GNO',      'INJ': 'INJ',       'KAR': 'KAR',
+        'KEEP': 'KEEP',    'KNC': 'KNC',       'LSK': 'LSK',
+        'LTP': 'LTP',      'LRC': 'LRC',       'MIR': 'MIR',
+        'OCEAN': 'OCEAN',  'PAXG': 'PAXG',     'RARI': 'RARI',
+        'REN': 'REN',      'XRP': 'XXRP',      'SRM': 'SRM',
+        'STORJ': 'STORJ',  'TBTC': 'TBTC',     'OGN': 'OGN',
+        'OXT': 'OXT'
+    }
```
Comment (shershah010) [inline harvest/api/kraken.py:49]: These are all the cryptos that kraken supports, if any new ones are added then this dictionary needs to be updated. This is to convert from the normal ticker name to the ticker that kraken uses in its api.

File: harvest/api/kraken.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+        return df
+
+    @API._exception_handler
+    def fetch_chain_info(self, symbol: str):
+        return {
+            "id": "n/a",
+            "exp_dates": [str_to_date(s) for s in self.watch_ticker[symbol].options],
+            "multiplier": 100
+        }
```
Comment (shershah010) [inline harvest/api/kraken.py:118]: What is `fetch_chain_info` supposed to do?

File: harvest/api/kraken.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+        df = df.rename(columns={"contractSymbol": "occ_symbol"})
+        df['exp_date'] = df.apply(lambda x: self.occ_to_data(x['occ_symbol'])[1], axis=1)
+        df = df[["occ_symbol", "exp_date", "strike", "type"]]
+        df.set_index('occ_symbol', inplace=True)
+
+        if not symbol in self.option_cache:
+            self.option_cache[symbol] = {}
+        self.option_cache[symbol][date] = df
+
+        return df
```
Comment (shershah010) [inline harvest/api/kraken.py:145]: What is `fetch_chain_data` supposed to do?

**Comment by shershah010** [comment]
I accidentally commited and pushed the build dir, I don't know how to revert. Also the tests don't run because of flask not being install properly.

File: harvest/api/kraken.py (hunk 3 above)
Comment (tfukaza) [reply harvest/api/kraken.py:118]: This function is for options, so no need to implement for cryptos

File: harvest/api/kraken.py (hunk 4 above)
Comment (tfukaza) [reply harvest/api/kraken.py:145]: This function also is mainly for options.

**Review by tfukaza — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -11,6 +11,7 @@
 
 # ignore generated files
 dist/
+build/
```
Comment (shershah010) [inline .gitignore:14]: Readded since this is a generated folder

File: setup.cfg [hunk 6]
```diff
@@ -21,6 +21,7 @@ install_requires =
     pyyaml
     tqdm
     pytz
+    tzlocal
```
Comment (shershah010) [inline setup.cfg:24]: Converts the system time into a timezone that pytz can use. I did not know a better way to do this without a package.

File: harvest/utils.py [hunk 7]
```diff
@@ -84,6 +86,14 @@ def mark_down(x):
 def has_timezone(date: dt.datetime) -> bool:
     return date.tzinfo is not None and date.tzinfo.utcoffset(date) is not None
 
+def set_system_timezone(date: dt.datetime) -> dt.datetime:
```
Comment (shershah010) [inline harvest/utils.py:89]: useful since the user will probably not set the timezone and they usually mean their timezone.

**Pull request merged**